### PR TITLE
minor fixes to ui thread sync for the desktop app and few other minor…

### PR DESCRIPTION
**Pull request by henrik-me**
… updates to obsolete references and first take on following the async methodology. Also added the AppStore build config to consistently build debug bits, the current one used a mix of debug and release bits which caused issues for my local builds. Not sure exactly what the idea is though with the build (@jmprieur, @jennyf19 @trwalke thinking you might be able to shed some light on that config). Did a few other things here and there. The UWP app and the desktop app and the unit tests now also runs on my box.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (jmprieur) [inline .vscode/launch.json]: @henrik-me : why do we need to commit this file?

**Comment by jmprieur** [comment]
LGTM. I don't understand all all the build configuration thing. Do we ever build MSAL for iOS in release now?

**Comment by henrik-me** [comment]
I realize there was some issues in the past regarding that. At this moment I do not have a Mac thus I can't validate this. Is there still a known issue around this? I would assume release should work.

File: .vscode/launch.json (hunk 1 above)
Comment (henrik-me) [reply .vscode/launch.json]: It is stuff auto generated for those who work out of VSCode. I can put it in gitignore if you think that is the right thing?

Comment (jmprieur) [reply .vscode/launch.json]: personally, I would ignore the .vscode and .vs folders (that's the guidance I've been giving for samples). Now if there are good reasons to keep them, you know better :)

**Comment by jmprieur** [comment]
maybe we should try to build this branch on a hosted Mac ?

**Comment by henrik-me** [comment]
Agree, need to figure out how I can effectively do that from my local box as well.

**Pull request merged**